homework/src/index.js

```diff
@@ -29,19 +32,119 @@
     });
     return elem;
   }
+  function renderRepositories(repository, container) {
+    container.innerHTML = '';                                        //what is this line for? cleaning each time it run old info
+    const table = createAndAppend('table', container);
+    const tbody = createAndAppend('tbody', table);
+    const tr1 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr1, {
+      text: 'Repository:'
+    });
+    const td12 = createAndAppend('td', tr1);
+    createAndAppend('a', td12, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name
+    });
+    const tr2 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr2, {
+      text: 'Description:'
+    });
+    createAndAppend('td', tr2, {
+      text: repository.description
+    });
+    const tr3 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr3, {
+      text: 'Forks:'
+    });
+    createAndAppend('td', tr3, {
+      text: repository.forks
+    });
+    const tr4 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr4, {
+      text: 'Updated:'
+    });
+    createAndAppend('td', tr4, {
+      text: repository.updated_at
+    });
+  }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function renderContributors(contributors, container) {
+    container.innerHTML = '';
+    createAndAppend('p', container, {
+      text: 'Contributions'
     });
+    const ul = createAndAppend('ul', container);
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul);
+      const a = createAndAppend('a', li, {
+        href: contributor.html_url,
+        target: '_blank',
+        text: contributor.login
+      });
+      createAndAppend('img', a, {
+        src: contributor.avatar_url
+      });
+      createAndAppend('div', a, {
+        text: contributor.contributions
+      });
+    });
+  }
+
+  function main2(err, repositories) {
```

Comment:
Your function `main2()` (can you think of a better name? (I guess you wouldn't call your second child `child2`, would you? 🙂) is called for two almost unrelated things:

1. Normal processing of the repositories.
2. Rendering errors (even by calling itself, recursively).

It would be more clear if you separated those two things. You could create a separate function for error handling:

```js
function handleError(err) {
  const root = document.getElementById('root');
  createAndAppend('div', root, { text: err.message, class: 'alert-error' });
}

function main2(repositories) {
  const root = document.getElementById('root');
  // ...
  fetchJSON(...)
    .then(...) {
      //...
    })
   .catch((err) => handleError(err));
}

function main1(url) {
  fetchJSON(url)
    .then((data) => main2(data))
    .catch((err) => handleError(err));
}

//...
```